app/cdap/components/Connections/Browser/ParsingConfigModal/store.tsx

```diff
@@ -125,34 +127,55 @@ export const performInitialLoad = (connection, sampleProperties, entity, dispatc
   );
   const sourcePluginName = sourcePluginNameProperty.description;
 
+  const cdapVersion = VersionStore.getState().version;
+
   let widgetKey;
   // Get the connection details in order to find the arttifact version
   // Assume the source plugin is in the same artifact as the connector
   const widgetObs = ConnectionsApi.getConnection(params).pipe(
     switchMap((connectionDetails: IConnectionDetails) => {
-      const connectorArtifact = connectionDetails.plugin.artifact;
+      const { artifact: connectorArtifact } = connectionDetails.plugin;
+      let pluginDetailsParams;
+      if (connectorArtifact.version) {
+        pluginDetailsParams = {
+          namespace,
+          artifactId: connectorArtifact.name,
+          version: connectorArtifact.version,
+          scope: connectorArtifact.scope,
+          extensionType: SOURCE_PLUGIN_TYPE,
+          pluginName: sourcePluginName,
+        };
+      } else {
+        // If the connection doesn't specify an artifact version,
```

Comment:
nit: does this comment still stand true?

Reply:
Yes, this is the improvement over #426 - the backend can give us the latest version and we don't need `semver` here.